platform/darwin/src/darwin_conversion.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <mbgl/util/logging.hpp>
+#include <mbgl/style/conversion.hpp>
+#include <mbgl/util/feature.hpp>
+#include <mbgl/util/optional.hpp>
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+
+// A minimal wrapper class conforming to the requirements for `objectMember(v, name)` (see mbgl/style/conversion.hpp)
+// This is necessary because using `NSObject*` as the value type in `optional<NSObject*>` causes problems for the ARC,
+// due to things like `optional(const value_type& __v)`
+class OptionalNSObjectValue {
+public:
+    OptionalNSObjectValue(NSObject *_value) : value(_value) {}
+    OptionalNSObjectValue() {}
+    
+    explicit operator bool() const {
+        return value;
+    }
+    
+    NSObject* operator*() {
+        assert(this);
```

Comment:
Use `NSCAssert()` instead of `assert()` for files that are part of the Objective-C targets. `NSCAssert()` provides better diagnostics via NSException and works with Xcode’s built-in Objective-C exception breakpoint.